Userland/Applications/Screenshot/ScreenshotWidget.cpp

```diff
@@ -0,0 +1,155 @@
+ */
+
+#include "ScreenshotWidget.h"
+#include "SelectableOverlay.h"
+#include <Applications/Screenshot/ScreenshotWindowGML.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/DateTime.h>
+#include <LibCore/File.h>
+#include <LibCore/StandardPaths.h>
+#include <LibCore/Timer.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/CheckBox.h>
+#include <LibGUI/Clipboard.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/Notification.h>
+#include <LibGUI/RadioButton.h>
+#include <LibGUI/SpinBox.h>
+#include <LibGUI/WindowServerConnection.h>
+#include <LibGfx/PNGWriter.h>
+#include <LibConfig/Client.h>
+#include <serenity.h>
+
+void ScreenshotWidget::timer_event(Core::TimerEvent&)
+{
+    stop_timer();
+
+    window()->close();
+
+    save_screenshot(false);
+}
+
+ScreenshotWidget::ScreenshotWidget()
+{
+    load_from_gml(screenshot_window_gml);
+
+    m_whole_button = *find_descendant_of_type_named<GUI::RadioButton>("wholescreen");
+    m_custom_button = *find_descendant_of_type_named<GUI::RadioButton>("customregion");
+    m_delay_spinbox = *find_descendant_of_type_named<GUI::SpinBox>("delay");
+    m_copy_checkbox = *find_descendant_of_type_named<GUI::CheckBox>("copy_to_clipboard");
+    m_cancel_button = *find_descendant_of_type_named<GUI::Button>("cancel_button");
+    m_ok_button = *find_descendant_of_type_named<GUI::Button>("ok_button");
+
+    m_window_sel = GUI::Window::construct();
+    m_overlay = m_window_sel->set_main_widget<SelectableOverlay>(m_window_sel);
+    m_overlay->callback = [&] {
+        save_screenshot(true);
+    };
+
+    m_window_sel->set_title("Screenshot");
+    m_window_sel->set_has_alpha_channel(true);
+    m_window_sel->set_fullscreen(true);
+
+    String type = Config::read_string("Screenshot", "General", "ScreenshotType", "Whole");
+    if (type == "Custom") {
+        m_whole_button->set_checked(false);
+        m_custom_button->set_checked(true);
+    } else {
+        m_whole_button->set_checked(true);
+        m_custom_button->set_checked(false);
+    }
+
+    m_delay_spinbox->set_value(Config::read_i32("Screenshot", "General", "Delay", 0));
+    m_copy_checkbox->set_checked(Config::read_bool("Screenshot", "General", "CopyToClipboard", true));
+
+    m_cancel_button->on_click = [&](auto) {
+        GUI::Application::the()->quit();
+    };
+
+    m_ok_button->on_click = [&](auto) {
+        String new_type = "Whole";
+        if (m_custom_button->is_checked()) {
+            new_type = "Custom";
+        }
+
+        Config::write_string("Screenshot", "General", "ScreenshotType", new_type);
+        Config::write_i32("Screenshot", "General", "Delay", m_delay_spinbox->value());
+        Config::write_bool("Screenshot", "General", "CopyToClipboard", m_copy_checkbox->is_checked());
+
+        m_output_path = Core::DateTime::now().to_string("screenshot-%Y-%m-%d-%H-%M-%S.png");
+
+        window()->set_frameless(true);
+        window()->set_opacity(0);
```

Comment:
One thing that was pointed out on Discord after I posted that gist is that the better way to keep the app open once the window is closed is to use the `Application::set_quit_when_last_window_deleted()` API. Then, you should be able to do `window()->close();` here instead of this weird opacity thing I came up with :)

So something like:
```c++
GUI::Application::the()->set_quit_when_last_window_deleted(false);
window->close();
```

Reply:
Then, in the timer callback, you should invoke this after saving the screenshot:
```c++
GUI::Application::the()->quit();
```